### ISSUE TITLE
Does defaultApplication override results with system properties?

### ISSUE THREAD

**Issue by arisc097** [typesafehub/config#408]
defaultReference does, but defaultApplication doesn't.

----------------------------------------
See what ConfigFactory.defaultApplication() returns:
----------------------------------------
Path simple-app.answer: 42
Path simple-lib.foo: This value comes from simple-app's application.conf
**Path simple-lib.whatever: This value comes from simple-app's application.conf**
Path simple-lib.hello: is not available
Path: simple-lib.hello, Problem:No setting at 'simple-lib.hello', expecting: string

----------------------------------------
See what ConfigFactory.defaultReference returns:
----------------------------------------
Path simple-lib.foo: This value comes from simple-lib's reference.conf
**Path simple-lib.whatever: This value comes from a system property**
Path simple-lib.hello: This value comes from simple-lib's reference.conf

**Comment by havocp** [comment]
Isn't there a defaultOverrides() with the properties?

**Comment by arisc097** [comment]
Yes you are right. I am just checking on what was the rationale behind this subtle difference that defaultReference honors system properties while defaultApplication doesn't.

**Comment by havocp** [comment]
These methods are used if you want to create your own custom alternative to ConfigFactory.load. By default you have defaultOverrides over defaultApplication over defaultReference. The reason system properties aren't in defaultApplication is that they are in defaultOverrides.
